tests/common/platform/interface_utils.py

```diff
@@ -125,27 +125,31 @@ def get_port_map(dut, asic_index=None):
 
     return port_mapping
 
-def get_physical_port_indices(duthost):
-    """Returns list of physical port numbers of the DUT"""
-    physical_port_indices = set()
+def get_physical_port_indices(duthost, logical_intfs=None):
+    """
+    @summary: Returns dictionary map of logical ports to corresponding physical port indices
+    @param logical_intfs: List of logical interfaces of the DUT
+    """
+    physical_port_index_dict = {}
 
-    intf_facts = duthost.interface_facts()['ansible_facts']['ansible_interface_facts']
-    phy_port = re.compile(r'^Ethernet\d+$')
-    phy_intfs = [k for k in intf_facts.keys() if re.match(phy_port, k)]
-    phy_intfs = natsorted(phy_intfs)
-    logging.info("physical interfaces = {}".format(phy_intfs))
+    if logical_intfs is None:
+        intf_facts = duthost.interface_facts()['ansible_facts']['ansible_interface_facts']
+        phy_port = re.compile(r'^Ethernet\d+$')
+        logical_intfs = [k for k in intf_facts.keys() if re.match(phy_port, k)]
+        logical_intfs = natsorted(logical_intfs)
+        logging.info("physical interfaces = {}".format(logical_intfs))
 
     for asic_index in duthost.get_frontend_asic_ids():
 	# Get interfaces of this asic
 	interface_list = get_port_map(duthost, asic_index)
-	interfaces_per_asic = {k:v for k, v in interface_list.items() if k in phy_intfs}
+	interfaces_per_asic = {k:v for k, v in interface_list.items() if k in logical_intfs}
```

Comment:
I had a chat with Guohan about this. I think the main issue is that we are running SFP test against disabled (admin down) ports. If we could skip the admin down ports, that would be a much easier and generic solution.

can you update PR with that approach?

Reply:
yes, we are passing all interfaces which are in conn_graph_facts that are based on the connection graph and will not include 'admin' down' port.  get_physical_port_indices() is a helper function which will give indices for the input list of logical interfaces. The list of interfaces is formed in setup() fixture in test_sfp.py